### PR TITLE
[Core] Avoid install python 3.11 conda

### LINKED ISSUE #2240: [Python 3.11] All remote image with python 3.11 is not supported
Due to the underlying ray issue https://github.com/ray-project/ray/issues/31606, the image with python 3.11 by default will fail. That indicates that TPU VM (has python 3.11) issue #2228  is still not fully resolved yet.

### PR DESCRIPTION
**Pull request by Michaelvll**
Partially fixes #2240, for the images without conda installed by default.

This PR also updates the tf version for tpu tests as the TPU no longer support the old version of Tensorflow https://cloud.google.com/tpu/docs/supported-tpu-configurations#tensorflow_2

Confirmed that this PR fixes the TPU VM issue, as the original image of the TPU VM does not have conda installed by default.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] All smoke tests: `pytest tests/test_smoke.py` 
- [x] All smoke tests: `pytest tests/test_smoke.py --aws`

### CODE REVIEW

File: sky/skylet/constants.py [hunk 1]
```diff
@@ -46,3 +46,13 @@
 #
 # Port on the remote spot controller that the dashboard is running on.
 SPOT_DASHBOARD_REMOTE_PORT = 5000
+
+# Install conda on the remote cluster if it is not already installed.
+# We do not install the latest conda with python 3.11 because ray has not
+# officially supported it yet.
```
Comment (concretevitamin) [inline sky/skylet/constants.py:52]: Nit: link to the ray issue?

File: tests/test_smoke.py [hunk 2]
```diff
@@ -1506,20 +1509,26 @@ def _get_cancel_task_with_cloud(name, cloud, timeout=15 * 60):
 
 # ---------- Testing `sky cancel` ----------
 @pytest.mark.aws
+@pytest.mark.skip(
```
Comment (concretevitamin) [inline tests/test_smoke.py:1512]: Hmm, it seems a bit dangerous to skip these core tests? Maybe we should endure the flakiness for now?

File: sky/templates/aws-ray.yml.j2 [hunk 3]
```diff
@@ -190,7 +190,7 @@ setup_commands:
     pip3 --version > /dev/null 2>&1 || (curl -sSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 get-pip.py && echo "PATH=$HOME/.local/bin:$PATH" >> ~/.bashrc);
     (type -a python | grep -q python3) || echo 'alias python=python3' >> ~/.bashrc;
     (type -a pip | grep -q pip3) || echo 'alias pip=pip3' >> ~/.bashrc;
-    (which conda > /dev/null 2>&1 && conda init > /dev/null) || (wget -nc https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && bash Miniconda3-latest-Linux-x86_64.sh -b && eval "$(~/miniconda3/bin/conda shell.bash hook)" && conda init && conda config --set auto_activate_base true);
```
Comment (concretevitamin) [inline sky/templates/aws-ray.yml.j2:193]: These couple of j2 seem to have 2 - 3 versions. (e.g., some have `conda init > /dev/null`, some don't). I guess if smoke tests all passed (does it by default cover all tests?) it's ok to use 1 version?

**Review by concretevitamin — COMMENTED** [review]
Thanks for identifying the issue @Michaelvll! Left a question about skipping some smoke tests.

File: tests/test_smoke.py (hunk 2 above)
Comment (Michaelvll) [reply tests/test_smoke.py:1512]: Should we instead fix #2207 ? It seems to be problematic tests.

File: sky/templates/aws-ray.yml.j2 (hunk 3 above)
Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:193]: It should be fine. The only difference between the two versions is just having `conda init` after the `which conda` at the beginning. Theoretically, it should not affect anything. We now have the `conda init` everywhere, and tested with `pytest tests/test_smoke.py` (on GCP) which indicates adding the `conda init` indeed does not cause problem.

File: sky/skylet/constants.py (hunk 1 above)
Comment (Michaelvll) [reply sky/skylet/constants.py:52]: Good point! Added. Thanks!

**Review by concretevitamin — APPROVED** [review]

File: tests/test_smoke.py (hunk 2 above)
Comment (concretevitamin) [reply tests/test_smoke.py:1512]: should we mark it as P0? To avoid skipping these core tests for too long.

**Pull request merged**